File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int m = matrix.length, n = matrix[0].length;
        int row = 0, coln = n - 1;
        while(row < m && coln >= 0){
            if(matrix[row][coln] == target) return true;
            else if(matrix[row][coln] > target) coln--;
            else row++;
        }
        return false;
    }
}

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int n = matrix[0].size(); 
        int m = matrix.size();
        int row = 0, col = n - 1;

        while(row < m && col >= 0){
            if(target == matrix[row][col]) return true;
            else if(target < matrix[row][col]) col--;
            else row++;    
        }
        return false;
    }
};
